flink-runtime/src/main/scala/org/apache/flink/runtime/jobmanager/JobManager.scala

```diff
@@ -1487,7 +1487,7 @@ class JobManager(
           }
         }
 
-        eg.fail(cause)
+        eg.cancel()
```

Comment:
I liked it that the job was not marked as "canceled()" (which gives no reason), but as failed, with a cause. The cause also shows up in the TaskManager logs, when the Task was "externally failed".

Any specific reason that this changes? Is this currently inconsistent?